Source/Dafny/DafnyMain.cs

```diff
@@ -28,14 +29,19 @@ public class DafnyFile {
       }
       return sourceFiles;
     }
-
     public DafnyFile(string filePath) {
       FilePath = filePath;
       BaseName = Path.GetFileName(filePath);
 
       var extension = Path.GetExtension(filePath);
       if (extension != null) { extension = extension.ToLower(); }
 
+      // Normalizing symbolic links appears to be not
+      // supported in .Net APIs, because it is very difficult in general
+      // So we will just use the absolute path, lowercased for all file systems.
+      // cf. IncludeComparer.CompareTo
+      CanonicalPath = Path.GetFullPath(filePath).ToLower();
```

Comment:
I think we should _not_ do the `.ToLower()`. Here's why:

* If we keep case and files `A.dfy` and `a.dfy` are listed (either on the command line or via `include` directives), then these will appear to be 2 different files. That would be as expected on a case-sensitive file system. On a case-insensitive file system, this would generate an error, but the fix is easy, which is for the user to consistently name this one file.

* If we convert to lower-case and files `A.dfy` and `a.dfy` are listed, then these will appear to be synonyms for one file, so one of them will be silently ignored. On a case-insensitive file system, this is good and expected. However, on a case-sensitive system, one of the files will be silently ignored and there's no workaround to get both files to be included.

Of these, I much prefer keeping the case.

Reply:
In the first round I mimicked how include files were handled. With this design choice, I
refactored to pull all the canonicalization of file paths (both include and command-line) into one method and used it uniformly.